### PR TITLE
BAN-166: Write component tests

### PR DESCRIPTION
**Pull request by chavlji**
[JIRA ticket](https://reciprocitylabs.atlassian.net/browse/BAN-166)

### CODE REVIEW

File: src/components/helpers/jest.ts [hunk 1]
```diff
@@ -1,8 +1,35 @@
...
+
+export async function propReflectsInAttributes(page: SpecPage, props: Record<string, unknown>): Promise<boolean> {
+  // Use this helper to test if all supported props are reflected in attributes
+  for (const [key, value] of Object.entries(props)) {
+    page.rootInstance[key] = value;
+    await page.waitForChanges();
+    if (page.root.getAttribute(kebabCase(key)) === null) return false;
+  }
+  return true;
+}
```
Comment (f-jauregui) [inline src/components/helpers/jest.ts:35]: Nice! 👍

File: src/components/zen-animate/test/zen-animate.spec.tsx [hunk 2]
```diff
@@ -1,4 +1,12 @@
 import { newSpecPage } from '@stencil/core/testing';
+import { htmlToElement } from '../../helpers/jest';
+
+const fakeSlot = htmlToElement('<div></div>')[0];
+
+import * as helpers from '../../helpers/helpers';
+helpers.getSlotElement = jest.fn(() => fakeSlot);
+helpers.waitNextFrame = jest.fn(() => new Promise(resolve => resolve(true)));
+
```
Comment (f-jauregui) [inline src/components/zen-animate/test/zen-animate.spec.tsx:9]: Please add test for the show property.

File: src/components/helpers/helpers.ts [hunk 3]
```diff
@@ -16,7 +12,7 @@ export function slotPassed(host: HTMLElement, slotName: string): boolean {
   return !!host.querySelector(selector);
 }
 
-export function getSlotElement(host: HTMLElement, slotName: string): HTMLElement | undefined {
+export function getSlotElement(host: HTMLElement, slotName?: string): HTMLElement | undefined {
```
Comment (f-jauregui) [inline src/components/helpers/helpers.ts:15]: This is a cool helper! 👍

File: src/components/zen-dropdown/test/zen-dropdown.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+describe('Zen-dropdown', () => {
+  it('renders', async () => {
+    const page = await newSpecPage({
+      components: [ZenDropdown],
+      html: `<zen-dropdown />`,
+    });
+    const dropdown = page.root;
+    dropdown.shadowRoot.querySelector('style').remove();
+    dropdown.shadowRoot.querySelectorAll('svg').forEach(svg => svg.remove());
+    expect(page.root).toEqualHtml(`
```
Comment (f-jauregui) [inline src/components/zen-dropdown/test/zen-dropdown.spec.tsx:25]: Please remove snapshot test if it's not necessary.

File: src/components/zen-dropdown/test/zen-dropdown.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+    dropdown = page.root;
+    options = htmlToElement(`<zen-option value="admin">Administrator</zen-option>
+        <zen-option value="reader">Reader</zen-option>
+        <zen-option value="contributor">Contributor</zen-option>`);
+
+    field = dropdown.shadowRoot.querySelector('.field');
+    simulateMouse('mousedown', field);
+    await page.waitForChanges();
+    jest.advanceTimersByTime(100);
+  });
```
Comment (f-jauregui) [inline src/components/zen-dropdown/test/zen-dropdown.spec.tsx:77]: Good idea here! 🙌

File: src/components/zen-menu-item/test/zen-option.spec.tsx [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+import { ZenOption } from '../zen-option';
+
+describe('Zen-option', () => {
+  it('renders', async () => {
+    const page = await newSpecPage({
+      components: [ZenOption],
+      html: `<zen-option value="Graham" />`,
+    });
+    expect(page.root).toEqualHtml(`<zen-option value="Graham">
```
Comment (f-jauregui) [inline src/components/zen-menu-item/test/zen-option.spec.tsx:12]: Please remove snapshot test if it's not necessary.

File: src/components/zen-menu-item/test/zen-option.spec.tsx [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+      html: `<zen-option />`,
+    });
+    const props = {
+      selected: true,
+      focused: true,
+      value: 'some value',
+      noHover: true,
+      disabled: true,
+    };
+    expect(await propReflectsInAttributes(page, props)).toBeTruthy();
```
Comment (f-jauregui) [inline src/components/zen-menu-item/test/zen-option.spec.tsx:35]: Loved this.

File: src/components/zen-tabs/test/zen-tabs.spec.tsx [hunk 8]
```diff
@@ -0,0 +1,51 @@
+import { newSpecPage, SpecPage } from '@stencil/core/testing';
+// import { propReflectsInAttributes } from '../../helpers/jest';
```
Comment (f-jauregui) [inline src/components/zen-tabs/test/zen-tabs.spec.tsx:2]: Please remove.

File: src/components/zen-tabs/test/zen-tabs.spec.tsx [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+  it('renders', async () => {
+    const page = await newSpecPage({
+      components: [ZenTabs],
+      html: `<zen-tabs />`,
+    });
+    expect(page.root).toEqualHtml(`<zen-tabs>
+      <mock:shadow-root></mock:shadow-root>
+    </zen-tabs>`);
+  });
+});
```
Comment (f-jauregui) [inline src/components/zen-tabs/test/zen-tabs.spec.tsx:16]: Please remove snapshot test if it's not necessary.

**Review by f-jauregui — CHANGES_REQUESTED** [review]
Awesome work, Jan!
Just some little updates.

Regarding snapshot tests, I don't know if they are necessary to achieve certain coverage, so I just commented to remove them **if necessary**.

File: src/components/zen-dropdown/test/zen-dropdown.spec.tsx (hunk 4 above)
Comment (chavlji) [reply src/components/zen-dropdown/test/zen-dropdown.spec.tsx:25]: Good one! 🙌🏻

**Review by mplesec — APPROVED** [review]
Good work 👍

**Review by f-jauregui — APPROVED** [review]
Well done!

**Pull request merged**